src/logic/process-logs.js

```diff
@@ -8,14 +8,15 @@ const loggly = require('./services/loggly');
 const datadog = require('./services/datadog');
 
 const requestLogRegex = new RegExp([
-  /^REPORT RequestId: ([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\t/,
-  /Duration: (\d+.\d+) ms\t/,
-  /Billed Duration: (\d+) ms\s?\t/,
-  /Memory Size: (\d+) MB\t/,
-  /Max Memory Used: (\d+) MB\t/,
-  /(?:Init Duration: (\d+.\d+) ms\t)?\n/,
+  /^REPORT RequestId: (?<requestId>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\t/,
+  /Duration: (?<duration>\d+.\d+) ms\t/,
+  /Billed Duration: (?<billedDuration>\d+) ms\s?\t/,
+  /Memory Size: (?<maxMemory>\d+) MB\t/,
+  /Max Memory Used: (?<memory>\d+) MB\t/,
+  /(?:Init Duration: (?<initDuration>\d+.\d+) ms\t)?\n/,
   // https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html
-  /(?:XRAY TraceId: (\d+-[0-9a-f]{8}-[0-9a-f]{24})\tSegmentId: ([0-9a-f]{16})\tSampled: (true|false)\t\n)?$/
+  // eslint-disable-next-line max-len
+  /(?:XRAY TraceId: (?<traceId>\d+-[0-9a-f]{8}-[0-9a-f]{24})\tSegmentId: (?<segmentId>[0-9a-f]{16})\tSampled: (?<sampled>true|false)\t\n)?$/
```

Comment:
Should we split this line up?

Reply:
yes please

Reply:
We want to make it clear that this whole group is considered optional, so when you split this up @krackerman , do it like:
```suggestion
  ...[
    /(?:XRAY TraceId: (?<traceId>\d+-[0-9a-f]{8}-[0-9a-f]{24})\t/,
    /SegmentId: (?<segmentId>[0-9a-f]{16})\t/,
    /...\n)?/
  ]
```